### PR TITLE
perf: improve performance of `_get_unmapped_api_elements_as_string`

### PR DESCRIPTION
**Pull request by lars-reimann**
### Summary of Changes

Reduce the runtime of the migration script on the API data for the `library-analyzer` itself from 12.1s to 7.7s. For larger libraries the improvement should be more pronounced still.

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.1) status: ✅ [SUCCESS](https://github.com/Safe-DS/Library-Analyzer/actions/runs/4800566006)

| Descriptor  |                                 Linter                                 |Files|Fixed|Errors|Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.1/descriptors/python_black)          |    4|    0|     0|0.8s        |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.1/descriptors/python_mypy)            |    4|     |     0|2.78s       |
|✅ PYTHON    |[ruff](https://megalinter.io/6.22.1/descriptors/python_ruff)            |    4|    0|     0|0.05s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.1/descriptors/repository_git_diff)|yes  |     |no    |0.03s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Library-Analyzer/actions/runs/4800566006)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

**Comment by lars-reimann** [comment]
@Aclrian It would be great if you could review this, if you have some time.

File: src/library_analyzer/processing/migration/_migrate.py [hunk 1]
```diff
@@ -146,99 +146,45 @@ def print_api_element(api_element: Attribute | Class | Function | Parameter | Re
...
-            if not is_included(result):
-                not_mapped_api_elements.append(function.id + "/" + result.name)
-        return not_mapped_api_elements
+        return [element for element in api_elements if element not in mapped_api_elements]
 
     def print(self, apiv1: API, apiv2: API) -> None:
         print("**Similarity**|**APIV1**|**APIV2**|**comment**\n:-----:|:-----:|:-----:|:----:|")
         table_body = self._get_mappings_for_table()
-        table_body.extend(self._get_not_mapped_api_elements_for_table(apiv1, apiv2))
+        table_body.extend(self._get_unmapped_api_elements_for_table(apiv1, apiv2))
```
Comment (Aclrian) [inline src/library_analyzer/processing/migration/_migrate.py:187]: This line can be removed to additionally increase performance.

**Review by Aclrian — COMMENTED** [review]
Everything else looks fine.

File: src/library_analyzer/processing/migration/_migrate.py [hunk 2]
```diff
@@ -146,99 +146,45 @@ def print_api_element(api_element: Attribute | Class | Function | Parameter | Re
...
         table_body.sort(key=lambda row: max(len(cell.split("/")) for cell in row.split("|")[:-1]))
```
Comment (Aclrian) [inline src/library_analyzer/processing/migration/_migrate.py:188]: This line can be removed to additionally increase performance.

Comment (lars-reimann) [reply src/library_analyzer/processing/migration/_migrate.py]: Thanks, I've removed the sorting.

**Pull request merged**